f5/bigip/contexts.py

```diff
@@ -0,0 +1,61 @@
+# coding=utf-8
+#
+# Copyright 2014 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+class TransactionContext(object):
+    def __init__(self, transaction):
+        self.transaction = transaction
+        self.bigip = transaction._meta_data['bigip']
+        self.icr = self.bigip._meta_data['icr_session']
+
+    def __enter__(self):
+        """Begins a new transaction context
+
+        When a transaction begins, this method will automatically be called
+        to set up the transaction.
+
+        Transaction IDs are automatically retrieved for you and the
+        appropriate headers are set so that future operations will reference
+        the transaction.
+
+        Headers are preserved so that after you exit the transaction, you will
+        be able to use the API object as you normally would.
+        """
+        self.transaction.create(json={})
```

Comment:
I get this when i do that
(ansible-2.0)SEA-ML-RUPP1:f5-common-python trupp$ python ok.py 
Traceback (most recent call last):
  File "ok.py", line 12, in <module>
    with TransactionContextManager(tx) as api:
  File "/Users/trupp/src/trupp/f5-common-python/f5/bigip/contexts.py", line 45, in __enter__
    self.transaction.create(empty)
TypeError: create() takes exactly 1 argument (2 given)